packages/data-explorer/src/main.ts

```diff
@@ -74,18 +74,29 @@ Vue.config.productionTip = false
 // This way the mainView can coordinate the events
 Vue.prototype.$eventBus = new Vue()
 
+const router = new Router({
+  base: process.env.NODE_ENV === 'production' ? packageJson.name : process.env.BASE_URL,
+  routes
+})
+
+// Navigation guard that makes sure that all explorer
+// query params are present.
+router.beforeEach((to, from, next) => {
+  globalThis.defaultRouteQuery = defaultRouteQuery
+  globalThis.to = to.query
+  if (to.name === 'de-view') {
+    const missingQueryParams = Object.keys(defaultRouteQuery).filter((i) => !Object.keys(to.query).includes(i))
+    if (missingQueryParams.length) {
+      next({ path: `/explorer/${to.params.entity}`, query: defaultRouteQuery })
+    } else { next() }
```

Comment:
why is this code needed , it looks like it trying to hide a issue that should be fixed at the source

Reply:
It verifies whether all query params are available when it enters the de-view route. Otherwise it redirects with all the required query params.

Reply:
so then it hides the error making it harder to fix, if a single param is missing , the whole thing now gets replaced with the default , this seems incorrect